### PR TITLE
common files added

### PR DESCRIPTION
**Pull request by jai38**

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
```
Comment (akartik80) [inline .idea/.gitignore]: Everything under `./idea` should go in gitignore

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,190 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>co.zecko.common.java</groupId>
```
Comment (akartik80) [inline pom.xml]:
```suggestion
    <groupId>co.zecko.common</groupId>
```

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    <artifactId>client</artifactId>
```
Comment (akartik80) [inline pom.xml]:
```suggestion
    <artifactId>public</artifactId>
```

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+    <version>0.0.1-SNAPSHOT</version>
+    <packaging>jar</packaging>
+
+    <name>Zecko Common Java</name>
```
Comment (akartik80) [inline pom.xml]:
```suggestion
    <name>Zecko Common Java Public</name>
```

File: target/apidocs/allclasses-index.html [hunk 5]
```diff
@@ -0,0 +1,179 @@
+<!DOCTYPE HTML>
```
Comment (akartik80) [inline target/apidocs/allclasses-index.html]: Why so many HTML classes?

File: src/main/java/co/zecko/common/java/pojo/Extensions.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import lombok.experimental.FieldDefaults;
+
+@Getter
+@Setter
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public class Extensions {
```
Comment (akartik80) [inline src/main/java/co/zecko/common/java/pojo/Extensions.java]: Where is this used?

**Review by akartik80 — CHANGES_REQUESTED** [review]
Every file other than `.java` files is junk, please remove

File: src/main/java/co/zecko/common/java/pojo/Extensions.java (hunk 6 above)
Comment (jai38) [reply src/main/java/co/zecko/common/java/pojo/Extensions.java]: in ErrorEntry and ErrorEntry is used in BaseEntry

File: target/apidocs/allclasses-index.html (hunk 5 above)
Comment (jai38) [reply target/apidocs/allclasses-index.html]: that has to be in .gitignore

File: .idea/.gitignore (hunk 1 above)
Comment (jai38) [reply .idea/.gitignore]: done

File: src/main/java/co/zecko/common/java/pojo/ErrorEntry.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+@NoArgsConstructor
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public class ErrorEntry implements Serializable {
+
+    private static final long serialVersionUID = -8307940712006208211L;
+
+    String message;
+
+    Extensions extensions;
```
Comment (akartik80) [inline src/main/java/co/zecko/common/java/pojo/ErrorEntry.java]: We don't need to show the cost to the client. This can be removed

**Review by akartik80 — CHANGES_REQUESTED** [review]

File: src/main/java/co/zecko/commonPublic/java/pojo/BaseDataBaseEntry.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+    private static final long serialVersionUID = 1349160665663941607L;
+
+    private Long id;
+
+    private Date createdAt;
+
+    private Date updatedAt;
+
+    private String createdBy;
```
Comment (akartik80) [inline src/main/java/co/zecko/commonPublic/java/pojo/BaseDataBaseEntry.java:28]: private is default and not required. You can change in next PR

**Review by akartik80 — APPROVED** [review]

**Pull request merged**